File: backend/schema/match.py
from datetime import datetime
from typing import Optional

from schema.base_schema import BaseSchema

class Match(BaseSchema):
    uuid: str
    user_id: str
    created_at: datetime
    finished_at: Optional[datetime]
    bot_A_uuid: str
    bot_B_uuid: str
    winner_uuid: Optional[str]

class MatchCreate(BaseSchema):
    bot_A_uuid: str
    bot_B_uuid: str

File: frontend/src/types/Match.ts
export type Match = {
    uuid: string;
    userId: string;
    createdAt: string | null;
    finishedAt: string | null;
    botAUuid: string | null;
    botBUuid: string | null;
    winnerBotId: string | null;
}

export type MatchCreate = {
    botAUuid: string;
    botBUuid: string;
};